core/src/main/java/org/apache/carbondata/core/datastore/chunk/store/ColumnPageWrapper.java

```diff
@@ -53,11 +60,75 @@ public int fillVector(int[] filteredRowId, ColumnVectorInfo[] vectorInfo, int ch
     throw new UnsupportedOperationException("internal error");
   }
 
-  @Override
-  public byte[] getChunkData(int rowId) {
-    return columnPage.getBytes(rowId);
+  @Override public byte[] getChunkData(int rowId) {
```

Comment:
complex type Metadata column also should be adaptive encoded.

Reply:
Currently, Struct and Array types are stored as Byte Array. But as they are storing in LV format it can be modified and fit in V format. This refactoring will be done as part of jira CARBONDATA-2605